src/coreclr/src/jit/codegenarm.cpp

```diff
@@ -998,13 +998,36 @@ void CodeGen::genCodeForStoreLclFld(GenTreeLclFld* tree)
     // Ensure that lclVar nodes are typed correctly.
     assert(!varDsc->lvNormalizeOnStore() || targetType == genActualType(varDsc->TypeGet()));
 
-    GenTree*    data = tree->gtOp1;
-    instruction ins  = ins_Store(targetType);
-    emitAttr    attr = emitTypeSize(targetType);
+    GenTree* data = tree->gtOp1;
+    emitAttr attr = emitTypeSize(targetType);
 
     assert(!data->isContained());
     genConsumeReg(data);
-    emit->emitIns_S_R(ins, attr, data->GetRegNum(), varNum, offset);
+    regNumber dataReg = data->GetRegNum();
+    if (varTypeIsFloating(targetType) && ((offset % emitTypeSize(tree)) != 0))
+    {
+        regNumber addr = tree->ExtractTempReg();
+        emit->emitIns_R_S(INS_lea, attr, addr, varNum, offset);
+        if (targetType == TYP_FLOAT)
+        {
+            regNumber floatAsInt = tree->GetSingleTempReg();
+            emit->emitIns_R_R(INS_vmov_f2i, EA_4BYTE, floatAsInt, dataReg);
```

Comment:
It could be improved a little if we distinguish small and large offsets. In the small case that could be encoded into `INS_str` and for the large offset case that can use `rsGetRsvdReg` register. I decided not to do that in this bug fix PR.